third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -127,6 +127,48 @@ impl ScriptCache {
     }
 }
 
+pub(crate) struct StructNameCache {
+    data: RwLock<(
+        BTreeMap<StructIdentifier, StructNameIndex>,
+        Vec<StructIdentifier>,
+    )>,
+}
+
+impl Clone for StructNameCache {
```

Comment:
So I was wondering why this needs a clone, realized it's because the loader has clone, out of curiosity, do you know why does the loader need the Clone trait?
 
(I was even asking because cloning a lock to obtain a new one feels a bit unusual pattern, e.g. if we needed to just extract the contents we can have a dedicated method w.o. immediately putting it into another RwLock)